lib/utilities.js

```diff
@@ -61,7 +62,52 @@ function invertObject(obj, override) {
   return result;
 }
 
+/**
+ * Check if a given tag is a custom component.
+ *
+ * @see https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/isCustomComponent.js
+ * @param {string} tagName - The name of the html tag.
+ * @param {Object} props   - The props being passed to the element.
+ * @return {boolean}
+ */
+function isCustomComponent(tagName, props) {
+  if (tagName.indexOf('-') === -1) {
+    return props && typeof props.is === 'string';
+  }
+  switch (tagName) {
+    // These are reserved SVG and MathML elements.
+    // We don't mind this whitelist too much because we expect it to never grow.
+    // The alternative is to track the namespace in a few places which is convoluted.
+    // https://w3c.github.io/webcomponents/spec/custom/#custom-elements-core-concepts
+    case 'annotation-xml':
+    case 'color-profile':
+    case 'font-face':
+    case 'font-face-src':
+    case 'font-face-uri':
+    case 'font-face-format':
+    case 'font-face-name':
+    case 'missing-glyph':
+      return false;
+    default:
+      return true;
+  }
+}
+
+/**
+ * Check if the installed React version supports setting unknown attributes
+ * on elements.
+ *
+ * @return {boolean}
+ */
+function reactSupportsUnknownAttributes() {
```

Comment:
**perf:** can we memoize the value here to minimize execution cycles?

```js
var PRESERVE_CUSTOM_ATTRS = React.version.split('.')[0] >= 16;
// JS will coerce the value to number due to the comparison
// http://www.ecma-international.org/ecma-262/5.1/#sec-11.8.5
```

**refactor:** can we rename the value as `PRESERVE_CUSTOM_ATTRS` (so it follows the convention of a constant) and add a comment to the [blog post](https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html) above the line?

The only disadvantage of this approach is that the value isn't readonly or a constant so it's possible for others to override it